ADAL/src/ADUserInformation.m

```diff
@@ -259,6 +259,35 @@ - (id)copyWithZone:(NSZone *)zone
     return info;
 }
 
+- (BOOL)isEqual:(id)object
+{
+    if (!object)
+    {
+        return NO;
+    }
+    
+    if (self == object)
+    {
+        return YES;
+    }
+    
+    if (![object isKindOfClass:[ADUserInformation class]])
+    {
+        return NO;
+    }
+    
+    ADUserInformation *rhs = (ADUserInformation *)object;
+    
+    BOOL result = YES;
+    result &= [self.allClaims isEqualToDictionary:rhs.allClaims] || (self.allClaims == rhs.allClaims);
+    result &= [self.rawIdToken isEqual:rhs.rawIdToken] || (self.rawIdToken == rhs.rawIdToken);
```

Comment:
For this particular case yes, but lets think in another way: if we decide to inherit it and create ADMutableUserInformation, where user can change properties, then "isEqual" will not work properly. I think we should not "assume" that other properties are parsed from rawIdToken token here.